network/src/protocols/discovery/mod.rs

```diff
@@ -324,20 +324,17 @@ where
         // If a peer is previously unknown, or has a newer epoch number, we update its
         // corresponding entry in the map.
         let self_peer_id =
-            PeerId::try_from(self.self_note.get_peer_id()).expect("PeerId parsing fails");
+            PeerId::try_from(self.self_note.peer_id.clone()).expect("PeerId parsing fails");
         for note in remote_notes {
-            let peer_id = PeerId::try_from(note.get_peer_id()).expect("PeerId parsing fails");
-            let peer_info_bytes = note.get_signed_peer_info().get_peer_info();
-            let peer_info: PeerInfo =
-                protobuf::parse_from_bytes(peer_info_bytes).expect("PeerInfo parsing fails");
+            let peer_id = PeerId::try_from(note.peer_id.clone()).expect("PeerId parsing fails");
+            let peer_info_bytes = &note.signed_peer_info.as_ref().unwrap().peer_info;
```

Comment:
`unwrap` doesn't seem like the best choice here.  Why is prost forcing an Option<> on this reference?  It doesn't appear to be declared as optional.

Reply:
All message types in proto3 are optional, the old library sort of hides this fact

Reply:
Also all of the surrounding code unwraps so this case isn't bad IMO as the whole section of code would need to be rewritten to not unwrap and that is outside the scope of this PR

Reply:
![image](https://user-images.githubusercontent.com/22089200/65937578-2b916980-e3d5-11e9-9abf-0308cf65b4ce.png)

Reply:
@zamsden : It is safe to unwrap in this function because the validity of the message has already been verified. Although it would definitely be better to convert this to a different `VerifiedMessage` type (with access without unwrap) to convey that information.

Reply:
I agree it is safe, it is just a bit ugly ;)